src/CommunityPatch/Patches/Perks/Intelligence/Engineering/SiegeTooltipHelper.cs

```diff
@@ -46,6 +46,16 @@ public static void UpdateRangedDamageTooltip(List<TooltipProperty> tooltips, flo
       rangedDamageProperty.ValueLabel = amplifiedRangedDamage.ToString();
     }
     
+    public static void UpdateMaxHpTooltip(List<TooltipProperty> tooltips, float bonusHp)
+    {
+      var property = FindMaxHpTooltipProperty(tooltips);
+      if (property == null) return;
+
+      Double.TryParse(property.ValueLabel, out var currentHp);
+      var buffedHp = (int) (currentHp + bonusHp);
+      property.ValueLabel = buffedHp.ToString();
+    }
+    
     public static void AddPerkTooltip(List<TooltipProperty> tooltips, PerkObject perk, float bonusValue) {
       if (Math.Abs(bonusValue) < 0.05) return;
```

Comment:
can you comment on this? why 0.05?

Reply:
Oh sorry, it doesn't have to be 0.05. Since ReSharper was complaining about the float check of bonusValue == 0f, it suggested me to use Math.Abs. The '0.05' could be any number close to zero. I am going to change it

Reply:
You can check against epsilon

Reply:
I did the checking with epsilon, but for some reason it was not returning the expected boolean everytime

Reply:
Oh right. I see what you did instead, looks fine 👍